ludwig/models/predictor.py

```diff
@@ -257,14 +257,14 @@ def calculate_overall_stats(
         training_set_metadata
 ):
     overall_stats = {}
-    for output_feature in output_features:
-        of_name = output_feature.feature_name
+    for of_name, output_feature in output_features.items():
         if of_name not in overall_stats:
             overall_stats[of_name] = {}
         output_feature.calculate_overall_stats(
```

Comment:
I would modify this function, instead of passing `overall_stats[of_name]`, which is an empty dictionary, let's pass only predictions, targets and metadata. And then remove
```
if of_name not in overall_stats:
             overall_stats[of_name] = {}
```
And just do:
```
overall_stats[of_name] = output_feature.calculate_overall_stats(
  predictions[of_name],  # predictions
  dataset.get(of_name),  # target
   training_set_metadata[of_name]  #output feature metadata
)
```

This will require also changing the implementation in various output feature classes to return a dictionary instead of modifying a dictionary.

Reply:
Glad you pointed this out.  Because I was going to ask the question that the current approach has each feature's `calculate_overall_stats()` method by modifying the dictionary that is passed as a parameter instead of returning a dictionary. 
I prefer returning a dictionary.  I understand that I'll have to modify all the features's `calculate_overall_stats`.

Reply:
Resolved by PR #879.